src/data/move.ts

```diff
@@ -6482,84 +6471,400 @@ export class FirstMoveTypeAttr extends MoveEffectAttr {
   }
 }
 
-export class RandomMovesetMoveAttr extends OverrideMoveEffectAttr {
-  private enemyMoveset: boolean | null;
-
-  constructor(enemyMoveset?: boolean) {
-    super();
-
-    this.enemyMoveset = enemyMoveset!; // TODO: is this bang correct?
-  }
-
-  apply(user: Pokemon, target: Pokemon, move: Move, args: any[]): boolean {
-    const moveset = (!this.enemyMoveset ? user : target).getMoveset();
-    const moves = moveset.filter(m => !m?.getMove().hasFlag(MoveFlags.IGNORE_VIRTUAL));
-    if (moves.length) {
-      const move = moves[user.randSeedInt(moves.length)];
-      const moveIndex = moveset.findIndex(m => m?.moveId === move?.moveId);
-      const moveTargets = getMoveTargets(user, move?.moveId!); // TODO: is this bang correct?
-      if (!moveTargets.targets.length) {
-        return false;
-      }
-      let selectTargets: BattlerIndex[];
-      switch (true) {
-        case (moveTargets.multiple || moveTargets.targets.length === 1): {
-          selectTargets = moveTargets.targets;
-          break;
-        }
-        case (moveTargets.targets.indexOf(target.getBattlerIndex()) > -1): {
-          selectTargets = [ target.getBattlerIndex() ];
-          break;
-        }
-        default: {
-          moveTargets.targets.splice(moveTargets.targets.indexOf(user.getAlly().getBattlerIndex()));
-          selectTargets =  [ moveTargets.targets[user.randSeedInt(moveTargets.targets.length)] ];
-          break;
-        }
-      }
-      const targets = selectTargets;
-      user.getMoveQueue().push({ move: move?.moveId!, targets: targets, ignorePP: true }); // TODO: is this bang correct?
-      user.scene.unshiftPhase(new MovePhase(user.scene, user, targets, moveset[moveIndex]!, true)); // There's a PR to re-do the move(s) that use this Attr, gonna put `!` for now
-      return true;
+/**
+ * Attribute used to call a move.
+ * Used by other move attributes: {@linkcode RandomMoveAttr}, {@linkcode RandomMovesetMoveAttr}, {@linkcode CopyMoveAttr}
+ * @see {@linkcode apply} for move call
+ * @extends OverrideMoveEffectAttr
+ */
+export class CallMoveAttr extends OverrideMoveEffectAttr {
```

Comment:
```suggestion
class CallMoveAttr extends OverrideMoveEffectAttr {
```
Just noticed, this isn't imported to any other files so it doesn't need to be exported.